library/src/main/java/com/novoda/downloadmanager/lib/FileDownloadInfo.java

```diff
@@ -454,4 +454,48 @@ private Long getLong(String column) {
             return cursor.getLong(cursor.getColumnIndexOrThrow(column));
         }
     }
+
+    static final class ControlStatus {
+
+        private int control;
+        private int status;
+
+        public ControlStatus(int control, int status) {
+            this.control = control;
+            this.status = status;
+        }
+
+        public boolean isPaused() {
+            return control == DownloadsControl.CONTROL_PAUSED;
+        }
+
+        public boolean isCanceled() {
+            return status == DownloadStatus.CANCELED;
+        }
+
+        static final class Reader {
```

Comment:
could have a discussion and modify this pattern in the FileDownloadInfo as well, but I would prefer if we do it in another PR